src/app/shared/scroll-with-loading/scroll-with-loading.directive.ts

```diff
@@ -0,0 +1,35 @@
+import {Directive, ElementRef, EventEmitter, HostListener, Output} from '@angular/core';
+import {LoadDirection} from './load-direction';
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+})
+export class ScrollWithLoadingDirective {
+    @Output() loadData = new EventEmitter<LoadDirection>();
+
+    scrollPos = 0;
+
+    borderOffset = 100;
+
+    constructor(private readonly elementRef: ElementRef) {}
+
+    @HostListener('scroll')
+    OnScroll(): void {
+        const containerElement = this.elementRef.nativeElement;
+        const scrollTop = containerElement.scrollTop;
+        const scrollBottom =
+            containerElement.scrollHeight -
+            containerElement.scrollTop -
+            containerElement.clientHeight;
```

Comment:
Рекомендовал бы воспользоваться деструктуризацией для получения свойств объекта в константах
```suggestion
        const {scrollTop, scrollHeight, clientHeight} = this.elementRef.nativeElement;
        const scrollBottom = scrollHeight - scrollTop - clientHeight;
```